### ISSUE TITLE
code: test example repos

### ISSUE THREAD

**Issue by akosyakov** [gitpod-io/gitpod#3341]
Test all our example repos and make sure that they have nice setup with Code.

**Comment by apolopena** [comment]
May depend on https://github.com/gitpod-io/gitpod/issues/3309

**Comment by jankeromnes** [comment]
@akosyakov Is Monday, March 15th a good time to do another round of testing with Code, or should I wait for some changes/upgrades?

**Comment by akosyakov** [comment]
I'm going to upgrade again today and then have to bring it on staging. Let's sync on Wedndesday.

**Comment by jankeromnes** [comment]
Okay, I've tested the new Code on staging against our example repositories. It worked great! 🎉

But I also found a few issues:

- [ ] Preview has broken icons in Firefox Nightly on Mac OS https://github.com/gitpod-io/gitpod/issues/3474
- [ ] When a port becomes active, the notification's `Preview` button works, but the `Browser` button gets blocked by the pop-up blocker (async `window.open()`) -- Maybe we can replace the `Browser` button with a straight-up `<a href="<preview-url>" target="_blank">Browser</a>` in the notification?
- [ ] Only "dumb" (word-based) autocomplete & no symbol hovers at all in our Python / Go / Rust / Ruby examples (worked well for Java though) -- we need to add the "smartness" extensions into each example repo's `.gitpod.yml`

> Important! We should not use hash based identifiers in .gitpod.yml anymore we are going drop it in future, but proper syntax is `${publisher}.${name}`, i.e. `redhat.vscode-java` , the latest extension will be installed from Open VSX on the workspace startup.

Other issues ("won't fix" for now):

- The "Welcome" opens on every new workspace, which seems a bit excessive. Not sure whether we can do anything about it (VS Code treats browser + origin as a new machine and welcome a user for each new machine.)

**Comment by akosyakov** [comment]
@jankeromnes we should make sure that smartness works in all our example repos for VS Code, i.e. add missing extension to .gitpod.yml and set proper settings, as far as I understood issues like https://github.com/gitpod-io/gitpod/issues/959 will pop up again, right?

**Comment by akosyakov** [comment]
updated all example and test repos
